archive/qt_installer_framework.ksy

```diff
@@ -0,0 +1,225 @@
+  Alternatively, this file may be used under the terms of the GNU General Public License version 3 as published by the Free Software Foundation with exceptions as appearing in the file LICENSE.GPL3-EXCEPT included in the packaging of this file. Please review the following information to ensure the GNU General Public License requirements will be met: https://www.gnu.org/licenses/gpl-3.0.html.
+  
+
+doc-ref:
+  - https://wiki.qt.io/Qt-Installer-Framework
+  - https://github.com/qtproject/installer-framework/blob/master/src/libs/installer/binaryformat.cpp
+  - https://github.com/qtproject/installer-framework/blob/master/src/libs/installer/binarycontent.cpp
+  - https://github.com/qtproject/installer-framework/blob/master/src/libs/installer/binarycontent.h
+
+params:
+  - id: magic_cookie_offset
+    type: u8
+    -orig-id: endOfBinaryContent - 8
+    doc:  "In order to use this spec find the magic cookie [0xf8, 0x68, 0xd6, 0x99, 0x1c, 0x0a, 0x63, 0xc2] in the last MiB of file and set this param to its offset."
+instances:
+  end_of_binary_content:
+    value: magic_cookie_offset + sizeof<header::cookie_identifier>
+  header:
+    pos: end_of_binary_content - sizeof<header>
+    type: header
+types:
+  array:
+    seq:
+      - id: size
+        type: u8
+      - id: value
+        size: size
+  string:
+    seq:
+      - id: size
+        type: u8
+      - id: value
+        size: size
+        type: str
+  header:
+    seq:
+      - id: meta_resources_count
+        type: u8
+        -orig-id: metaResourcesCount
+      - id: unkn
+        type: u8
+        repeat: expr
+        repeat-expr: 2
+      - id: cookie
+        type: cookie_identifier
+        -orig-id: magicCookie
+
+    instances:
+      other_stuff_size:
+        value: 4 * sizeof<u8>
+        doc: "meta count, offset/length collection index, marker, cookie..."
+      binary_descriptor_offset:
+        value: _root.end_of_binary_content - other_stuff_size
+        -orig-id: posOfResourceCollectionsSegment
+      binary_descriptor:
+        io: _root._io
+        pos: binary_descriptor_offset
+        size: sizeof<binary_descriptor>
+        type: binary_descriptor
+
+    types:
+      marker_identifier:
+        seq:
+          - id: type
+            type: u1
+            enum: type
+          - id: signature
+            contents: [0x32, 0x02, 0x12]
+        enums:
+          type:
+            0x33: installer
+            0x34: uninstaller
+            0x35: updater
+            0x36: package_manager
+      
+      cookie_identifier:
+        seq:
+          - id: type
+            type: u1
+            enum: type
+          - id: signature
+            contents: [0x68, 0xd6, 0x99, 0x1c, 0x0a, 0x63, 0xc2]
+        enums:
+          type:
+            0xf8: binary
+            0xf9: data
+
+      range:
+        -orig-id: https://github.com/qtproject/installer-framework/blob/cf8d5b9a650cc413dde79cf7d72da407c8993ddc/src/libs/installer/fileio.cpp#L52
```

Comment:
It's unusual to have URL in `-orig-id`, why don't you use `doc-ref` instead?

Reply:
Nice catch.

Reply:
Fixed.